### PR TITLE
Flattener Patch

### LINKED ISSUE #4510: Flatten Code not generated if, compile config file contains remappings
## Problem 
A smart contract doesn't create the expected flattened code if the contract is contains remappings. 

## Steps to reproduce
Add the following `openzeppelin` remapping to the contract 
```json
{
  "language": "Solidity",
  "settings": {
      "optimizer": {
          "enabled": true,
          "runs": 200
    },
    "outputSelection": {
          "*": {
          "": ["ast"],
          "*": ["abi", "metadata", "devdoc", "userdoc", "storageLayout", "evm.legacyAssembly", "evm.bytecode", "evm.deployedBytecode", "evm.methodIdentifiers", "evm.gasEstimates", "evm.assembly"]
        }
    },
    "remappings":[
      "@openzeppelin/contracts/=https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.5.0/contracts/"
    ]
  }
}
``` 

Create contract HelloWorld

```solidity
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.7.0 <0.9.0;
import "@openzeppelin/contracts/access/Ownable.sol";
contract HelloWorld{
}
```
Compile the contract using the compiler config file and try to flatten it. it doesn't generate the expected flatten code but does compile the contract. 

If the import is replaced by exact url, it does produce `HelloWorld_flattened.sol` file
```solidity
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.7.0 <0.9.0;
import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.5.0/contracts/access/Ownable.sol";
contract HelloWorld{
}
```

### PR DESCRIPTION
**Pull request by joeizang**
fix #4510

### CODE REVIEW

File: libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 const IMPORT_SOLIDITY_REGEX = /^\s*import(\s+).*$/gm;
 const SPDX_SOLIDITY_REGEX = /^\s*\/\/ SPDX-License-Identifier:.*$/gm;
 
-export function getDependencyGraph(ast, target) {
+export function getDependencyGraph(ast, target: string) {
   const graph = tsort();
+  console.log(graph, target, ast)
```
Comment (Aniket-Engg) [inline libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts:6]: remove log

File: libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts [hunk 2]
```diff
@@ -12,23 +13,25 @@ export function getDependencyGraph(ast, target) {
...
   let concat = '';
   for (const file of files) {
-    const source = sources[file].content;
+    const source = sources[file] && sources[file]['content'] && sources[file]['content'].length > 0 ? sources[file]['content'] : '';
     const sourceWithoutImport = source.replace(IMPORT_SOLIDITY_REGEX, '');
     const sourceWithoutSPDX = sourceWithoutImport.replace(SPDX_SOLIDITY_REGEX, '');
     concat += `\n// File: ${file}\n\n`;
     concat += sourceWithoutSPDX;
   }
+  console.log(concat)
```
Comment (Aniket-Engg) [inline libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts:22]: same here

File: libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts [hunk 3]
```diff
@@ -12,23 +13,25 @@ export function getDependencyGraph(ast, target) {
...
   return concat;
 }
 
 function _traverse(graph, visited, ast, name) {
   let currentAst = null
-  currentAst = ast[name].ast
+  console.log(ast)
```
Comment (Aniket-Engg) [inline libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts:28]: here

File: libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts:22]: removed

File: libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts (hunk 1 above)
Comment (joeizang) [reply libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts:6]: removed

File: libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts (hunk 3 above)
Comment (joeizang) [reply libs/remix-ui/solidity-compiler/src/lib/logic/flattenerUtilities.ts:28]: removed

**Review by Aniket-Engg — COMMENTED** [review]
It seems it doesn't process the nested imports

**Comment by yann300** [comment]
pr https://github.com/ethereum/remix-project/pull/4530

**Pull request closed without merging**